gammapy/data/data_store.py

```diff
@@ -284,6 +284,43 @@ def load_many(self, obs_ids, hdu_type=None, hdu_class=None):
 
         return things
 
+    def check_observations(self):
+        """Perform some sanity checks for all observations.
+
+        Returns
+        -------
+        OrderedDict : dictionary containing failure messages for all runs that fail a check.
+        """
+
+        results = OrderedDict()
+        nfail = 0
+
+        # Loop over all obs_ids in obs_table
+        for obs_id in self.obs_table['OBS_ID']:
+            messages = []
+            # Check that events table is not empty
+            if len(self.obs(obs_id).events.table) == 0:
+                messages.append('events table empty')
+            # Check that thresholds are meaningful for aeff
+            if self.obs(obs_id).aeff.meta['LO_THRES'] >= self.obs(obs_id).aeff.meta['HI_THRES']:
+                messages.append('LO_THRES >= HI_THRES in effective area meta data')
+            # Check that maximum value of aeff is greater than zero
+            if np.max(self.obs(obs_id).aeff.data.data) <= 0:
+                messages.append('maximum entry of effective area table <= 0')
+            # Check that maximum value of edisp matrix is greater than zero
+            if np.max(self.obs(obs_id).edisp.data.data) <= 0:
+                messages.append('maximum entry of energy dispersion table <= 0')
+            # Check that thresholds are meaningful for psf
+            if self.obs(obs_id).psf.energy_thresh_lo >= self.obs(obs_id).psf.energy_thresh_hi:
+                messages.append('LO_THRES >= HI_THRES in psf meta data')
+            if len(messages) > 0:
+                results[obs_id] = messages
+                nfail += 1
+
+        print('Checks failed for {:d} out of {:d} observations'.format(nfail, len(self.obs_table['OBS_ID'])))
```

Comment:
If you `print` here, then it's very hard for callers to suppress (e.g. from pipelines).
I would suggest you leave this print to the caller, or if you want to do it from Gammapy, make a separate methods where you print the report.